test/TestHelpers.jl

```diff
@@ -198,6 +198,24 @@ end
     val
 end
 
+@inline function Base.deleteat!(A::OffsetArray, i::Int)
+    checkbounds(A, i)
```

Comment:
@nalimilan I don't think it makes sense to add `@boundscheck` annotations in one place but not the other very similar cases in the same file for the same type. I think those changes would be best made all at once in a separate PR.